src/models/yamlProjectConfiguration.ts

```diff
@@ -318,146 +285,171 @@ export class YamlProjectConfiguration {
-        }
-        if (
-            configurationFileContent.plugins?.sdkGenerator &&
-            !Array.isArray(configurationFileContent.plugins?.sdkGenerator)
-        ) {
-            throw new Error("sdkGenerator must be an array");
-        }
-
-        const plugins: YamlPluginsConfiguration | undefined = configurationFileContent.plugins;
-
-        if (configurationFileContent.cloudProvider) {
-            if (!cloudProviders.includes(configurationFileContent.cloudProvider)) {
-                throw new Error(
-                    `The cloud provider ${configurationFileContent.cloudProvider} is invalid. Please use ${CloudProviderIdentifier.GENEZIO} or ${CloudProviderIdentifier.SELF_HOSTED_AWS}.`,
-                );
-            }
-        }
-
-        const scripts: YamlScriptsConfiguration | undefined = configurationFileContent.scripts;
+    static async create(
+        configurationFileContent: RawYamlConfiguration,
+    ): Promise<YamlProjectConfiguration> {
+        const methodConfigurationSchema = zod
+            .object({
+                name: zod.string(),
+                type: zod.nativeEnum(TriggerType).optional(),
+                cronString: zod.string().optional(),
+            })
+            .refine(({ type, cronString }) => {
+                if (type === TriggerType.cron && cronString === undefined) return false;
+
+                return true;
+            }, "Cron methods must have a cronString property.")
+            .refine(({ type, cronString }) => {
+                if (type === TriggerType.cron && cronString && !isValidCron(cronString)) {
+                    return false;
+                }
+
+                return true;
+            }, "The cronString is not valid. Check https://crontab.guru/ for more information.")
+            .refine(({ type, cronString }) => {
+                const cronParts = cronString?.split(" ");
+                if (
+                    type === TriggerType.cron &&
+                    cronParts &&
+                    cronParts[2] != "*" &&
+                    cronParts[4] != "*"
+                ) {
+                    return false;
+                }
+
+                return true;
+            }, "The day of the month and day of the week cannot be specified at the same time.");
+
+        const configurationFileSchema = zod.object({
+            name: zod.string().refine((value) => {
+                const nameRegex = new RegExp("^[a-zA-Z][-a-zA-Z0-9]*$");
+                if (!nameRegex.test(value)) {
+                    throw new Error(
+                        "The project name is not valid. It must start with a letter and contain only letters, numbers and dashes",
+                    );
+                }
+
+                return true;
+            }),
+            region: zod.enum(regions).default("us-east-1"),
+            language: zod.nativeEnum(Language).default(Language.ts),
+            cloudProvider: zod
```

Comment:
The error here states:
```
Field `cloudProvider`:
	- Invalid enum value. Expected 'aws' | 'genezio' | 'selfHostedAws' | 'capybara' | 'capybaraLinux', received 'dsjan'
```
There are two issues here:
- `capybara` and `capybaraLinux` are not yet documented for external use.
- `aws` is deprecated in favor of `genezio`.

It will be ideal to create a custom error message, otherwise we risk confusing users:
Something like:
```
Invalid value for cloudProvider. The supported values are `genezio` or `selfHostedAws`.
```

Note: The underlying implementation shouldn't stop us from using `capybara` or `capybaraLinux` because they are actually used for testing.